src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
+    private int capacity = DEFAULT_CAPACITY;
+    private int treshold = (int)(capacity * LOAD_FACTOR);
+    private Node[] table = new Node[capacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > treshold) {
+            resize();
+        }
+        for (Node element : table) { //To check if the element already exists
+            if (element != null && (element.getValue().equals(value)
+                    || element.getValue() == value)) {
+                return;
+            }
+            if (element != null && element.getNext() != null) {
+                while (element != null) {
+                    if (element.getValue() == value || element.getValue().equals(value)) {
+                        return;
+                    }
+                    element = element.getNext();
+                }
+            }
+        }
+        if (table[getIndexFromHash(key)] != null //This case we use when no nextNode
+                && table[getIndexFromHash(key)].getNext() == null) {
+            Node element = table[getIndexFromHash(key)];
+            if (element.getKey() == key) {
+                element.setValue(value);
+                return;
+            }
+            table[getIndexFromHash(key)]
+                    .setNext(new Node<>(getHash(key),key,value,null));
+            size++;
+            return;
+        }
+        if (table[getIndexFromHash(key)] != null //This case we use when we have nextNode
+                && table[getIndexFromHash(key)].getNext() != null) {
+            Node element = table[getIndexFromHash(key)];
+            while (element.getNext() != null) {
+                if (element.getKey() == key) {
+                    element.setValue(value);
+                    return;
+                }
+                element = element.getNext();
+            }
+            element.setNext(new Node<>(getHash(key),key,value,null));
+            size++;
+            return;
+        }
+        table[getIndexFromHash(key)] = new Node<>(getHash(key),key,value,null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndexFromHash(key);
+        Node<K, V> returnNode = table[index];
+        while (returnNode != null) {
+            if (Objects.equals(returnNode.getKey(), key)) {
+                return returnNode.getValue();
+            }
+            returnNode = returnNode.getNext();
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int h = (key.hashCode() * 3) * (key.hashCode() * 5) / 17;
+        if (h < 0) {
+            h = h * -1;
+            return h;
+        }
+        return h;
+    }
+
+    private int getIndexFromHash(K key) {
+        return key == null ? 0 : Math.abs(getHash(key) % table.length);
+    }
+
+    public void resize() {
```

Comment:
```suggestion
    private void resize() {
```

все методы, которые используються только внутри этого класса нужно делать приватными